config/confignet/confignet.go

```diff
@@ -30,14 +34,28 @@ type NetAddr struct {
 	// Transport to use. Known protocols are "tcp", "tcp4" (IPv4-only), "tcp6" (IPv6-only), "udp", "udp4" (IPv4-only),
 	// "udp6" (IPv6-only), "ip", "ip4" (IPv4-only), "ip6" (IPv6-only), "unix", "unixgram" and "unixpacket".
 	Transport string `mapstructure:"transport"`
+
+	ReusePort bool `mapstructure:"reuseport"`
 }
 
 func (na *NetAddr) Dial() (net.Conn, error) {
-	return net.Dial(na.Transport, na.Endpoint)
+	l := net.Dialer{
+		Control: control,
+	}
+	return l.Dial(na.Transport, na.Endpoint)
 }
 
 func (na *NetAddr) Listen() (net.Listener, error) {
-	return net.Listen(na.Transport, na.Endpoint)
+	l := net.ListenConfig{
+		Control: control,
+	}
+	return l.Listen(context.Background(), na.Transport, na.Endpoint)
+}
+
+func control(_ string, _ string, c syscall.RawConn) error {
+	return c.Control(func(fd uintptr) {
+		unix.SetsockoptInt(int(fd), unix.SOL_SOCKET, unix.SO_REUSEPORT, 1)
```

Comment:
I would assume we need a build tag + alternative implementation for Windows?

Reply:
Sure, everything is mentioned in the issue. This PR is just to reproduce my example in the issue

Reply:
I believe there is no alternative for windows. Probably, need to leave generic implementation for other systems